app/api/permissions/entitiesPermissions.ts

```diff
@@ -0,0 +1,93 @@
+import entities from 'api/entities/entities';
+import users from 'api/users/users';
+import userGroups from 'api/usergroups/userGroups';
+import { unique } from 'api/utils/filters';
+import { GroupMemberSchema, UserGroupSchema } from 'shared/types/userGroupType';
+import { EntitySchema } from 'shared/types/entityType';
+import { AccessLevels, PermissionType, MixedAccess } from 'shared/types/permissionSchema';
+import { PermissionSchema } from 'shared/types/permissionType';
+import { MemberWithPermission } from 'shared/types/entityPermisions';
+
+const setAdditionalData = (
+  peopleList: (GroupMemberSchema | UserGroupSchema)[],
+  permission: PermissionSchema,
+  additional: (data: GroupMemberSchema | UserGroupSchema) => {}
+) => {
+  const userData = peopleList.find(u => u._id!.toString() === permission._id.toString());
```

Comment:
there is a helper ```WithId``` that can take care of the "Object is possibly undefined" for the _id. its the same helper that models use to make sure _id its not optional.
https://github.com/huridocs/uwazi/blob/042696c744116b6e1cc52385d997c26f2dce3fde/app/api/odm/models.ts#L7-L9

you can type the params like 
```
  peopleList: (WithId<GroupMemberSchema> | WithId<UserGroupSchema>)[],
```

Reply:
@daneryl it's very nice to have this helper, however, when I did the change I had some typing errors in the function that uses this one because of a conflict between the passed types. Probably we are not using here enough strongly typed variables, but it seems not so easy to resolve the conflicts. What do you think about keeping the operator `!`?